### PR TITLE
issue #27477: add created, last updated columns and enhance postComment

### PR DESCRIPTION
**Pull request by garyhgohoos**

### CODE REVIEW

File: foundation-database/manifest.js [hunk 1]
```diff
@@ -1093,32 +1095,52 @@
...
+    "public/tables/curr_rate.sql",
+    "public/tables/custinfo.sql",
     "public/tables/docass.sql",
+    "public/tables/emp.sql",
+    "public/tables/incdt.sql",
+    "public/tables/invchead.sql",
     "public/tables/invcitem.sql",
-    "public/tables/ipshead.sql",
     "public/tables/ipsass.sql",
+    "public/tables/ipshead.sql",
```
Comment (gilmoskowitz) [inline foundation-database/manifest.js]: Please leave `ipshead.sql` before `ipsass.sql`

File: foundation-database/manifest.js [hunk 2]
```diff
@@ -1093,32 +1095,52 @@
...
+    "public/tables/quhead.sql",
     "public/tables/quitem.sql",
+    "public/tables/salesrep.sql",
     "public/tables/saletype.sql",
     "public/tables/shiphead.sql",
+    "public/tables/shiptoinfo.sql",
     "public/tables/source.sql",
     "public/tables/tax.sql",
-    "public/tables/taxass.sql",
+    "public/tables/taxauth.sql",
```
Comment (gilmoskowitz) [inline foundation-database/manifest.js]: Please restore `taxass.sql` to the manifest.

File: foundation-database/public/trigger_functions/bomhead.sql [hunk 3]
```diff
@@ -1,37 +1,79 @@
...
+ FOR EACH ROW EXECUTE PROCEDURE _bomheadBeforeTrigger();
+
+
+CREATE OR REPLACE FUNCTION _bomheadAfterTrigger() RETURNS TRIGGER AS $$
+-- Copyright (c) 1999-2014 by OpenMFG LLC, d/b/a xTuple.
+-- See www.xtuple.com/CPAL for the full text of the software license.
+DECLARE
+  _revid INTEGER;
+  _check TEXT;
+  _cmnttypeid INTEGER;
```
Comment (gilmoskowitz) [inline foundation-database/public/trigger_functions/bomhead.sql]: These 3 variables can be removed as they're not used.

**Comment by gilmoskowitz** [comment]
Very nice work. The code is cleaner than before as well has having new functionality.
+1 when CI passes.

The CI failure comes from an error loading the triggers into the database:
```
 psql:/home/travis/build/xtuple/xtuple/scripts/output/build_dev.sql:77097: ERROR:  record "new" has no field "itemsite_lastupdated"
CONTEXT:  PL/pgSQL function _itemsitetrigger() line 112 at assignment
SQL statement "update itemsite set itemsite_qtyonhand=(itemsite_qtyonhand + itemsite_nnqoh)"
PL/pgSQL function inline_code_block line 4 at SQL statement
```

**Comment by gilmoskowitz** [comment]
Now the CI failure appears to be a null value for `cohead_lastupdated` (full error below). That's probably the `else` cases in all of the triggers that choose whether to set `*_lastupdated` or `*_created`. The fix is probably to remove the `else` and move `end if` so `*_lastupdated` always gets set on insert and update.
```
2016-06-20 21:28:57 UTC WARNING:  Error: null value in column "cohead_lastupdated" violates not-null constraint
	    at Object.XT.Data.createRecord (eval at <anonymous> (js_init:920:20), <anonymous>:419:20)
	    at Object.XT.Data.commitRecord (eval at <anonymous> (js_init:920:20), <anonymous>:363:14)
	    at Object.XT.Rest.post (eval at <anonymous> (js_init:920:20), <anonymous>:99:14)
	    at post:35:19
	    at null.<anonymous> (post:49:2)
	
2016-06-20 21:28:57 UTC ERROR:  unhandledError
2016-06-20 21:28:57 UTC DETAIL:  js_init() LINE 372:       throw "unhandledError";
2016-06-20 21:28:57 UTC STATEMENT:  select xt.post($${ 
	  "nameSpace":"XM", 
	  "type":"SalesOrder", 
	  "data":{
	  "wasQuote": false,
	  "status": "O",
	  "saleType": "CUST",
	  "calculateFreight": true,
	  "margin": 7.68,
	  "subtotal": 9.8,
	  "taxTotal": 0.49,
	  "freight": 0,
	  "miscCharge": 0,
	  "total": 10.29,
	  "site": "WH1",
	  "currency": "USD",
	  "orderDate": "2014-10-06T22:28:16.844Z",
	  "customer": "8B293D0B-1DE7-4000-8F67-3BD6C4FD2280",
	  "terms": "2-10N30",
	  "salesRep": "A2AC258A-7AA9-4E30-E9F9-C3D5CBB670A5",
	  "taxZone": "VA TAX",
	  "billtoContact": "35",
	  "shipto": "a29656a6-2aeb-4c2b-a5eb-95a8eaaf19c8",
	  "shiptoContact": "35",
	  "shipZone": "DOMESTIC1",
	  "scheduleDate": "2014-10-06T00:00:00.000Z",
	  "packDate": "2014-10-06T00:00:00.000Z",
	  "commission": 0.075,
	  "freightWeight": 3.75,
	  "shipVia": "UPS-GROUND-UPS Ground",
	  "billtoName": "Tremendous Toys Incorporated",
	  "billtoContactFirstName": "Mike",
	  "billtoContactLastName": "Farley",
	  "billtoContactPhone": "703-931-4269",
	  "billtoContactFax": "703-931-2212",
	  "billtoAddress1": "Tremendous Toys Inc.",
	  "billtoAddress2": "101 Toys Place",
	  "billtoCity": "Walnut Hills",
	  "billtoState": "VA",
	  "billtoPostalCode": "22209",
	  "billtoCountry": "United States",
	  "shiptoName": "Olde Towne Store 1",
	  "shiptoContactFirstName": "Jake",
	  "shiptoContactLastName": "Sweet",
	  "shiptoContactPhone": "800-321-5433",
	  "shiptoContactFax": "703-931-2212",
	  "shiptoAddress1": "Olde Towne Toys Store 1",
	  "shiptoAddress2": "1 Duke Street",
	  "shiptoCity": "Alexandria",
	  "shiptoState": "VA",
	  "shiptoPostalCode": "22201",
	  "shiptoCountry": "United States",
	  "lineItems": [
	    {
	      "firm": false,
	      "quantityUnitRatio": 1,
	      "scheduleDate": "2014-10-06T00:00:00.000Z",
	      "subNumber": 0,
	      "status": "O",
	      "atShipping": 0,
	      "shipped": 0,
	      "item": "YTRUCK1",
	      "site": "WH1",
	      "lineNumber": 1,
	      "taxType": "Taxable",
	      "quantityUnit": "EA",
	      "priceUnit": "EA",
	      "unitCost": 2.1193,
	      "price": 9.8,
	      "extendedPrice": 9.8,
	      "tax": 0.49,
	      "discount": 0,
	      "listPriceDiscount": 0.10828,
	      "markup": 3.624168,
	      "listPrice": 10.99,
	      "priceUnitRatio": 1,
	      "quantity": 1,
	      "basePrice": 9.8,
	      "customerPrice": 9.8
	    }
	  ]
	}, 
	  "username":"admin" 
	}$$);
```

**Comment by garyhgohoos** [comment]
i did not specify "not null" on the _lastupdated columns that i added.  cohead_lastupdated already existed.  i only changed cohead

**Pull request merged**